GPU/Common/FramebufferManagerCommon.cpp

```diff
@@ -547,27 +547,24 @@ void FramebufferManagerCommon::SetDepthFrameBuffer(bool isClearingDepth) {
 		return;
 	}
 
+	// First time use of this framebuffer's depth buffer.
+	currentRenderVfb_->usageFlags |= FB_USAGE_RENDER_DEPTH;
+
 	// If this first draw call is anything other than a clear, "resolve" the depth buffer,
 	// by copying from any overlapping buffers with fresher content.
-	if (!isClearingDepth) {
+	if (!isClearingDepth && useBufferedRendering_) {
 		CopyToDepthFromOverlappingFramebuffers(currentRenderVfb_);
 
-		// Special compatibility trick for Burnout Dominator lens flares. Not sure how to best generalize this. See issue #11100
-		if (PSP_CoreParameter().compat.flags().UploadDepthForCLUTTextures && (currentRenderVfb_->usageFlags & FB_USAGE_CLUT) != 0) {
-			// Set the flag, then upload memory contents to depth channel.
+		// Special compatibility trick for Burnout Dominator lens flares. See issue #11100
+		if (PSP_CoreParameter().compat.flags().UploadDepthForCLUTTextures && currentRenderVfb_->z_address > 0x04110000) {
```

Comment:
What if we did this only when `vfb->lastFrameNewSize == gpuStats.numFlips` rather than based on the z_address?  Would that work?

We do PerformStencilUpload / PerformMemoryUpload when creating a framebuf, I don't think it'd be so awful to also upload depth, and might allow it to be more generic.

-[Unknown]

Reply:
Hm, I could try that. I guess since the Z buffer is static, only uploading it on create/resize would work. I previously tried uploading it on create and it didn't work, but I didn't realize then that it was because the resize later ruined it...